mm2src/coins/nft/storage/wasm/wasm_storage.rs

```diff
@@ -315,6 +369,60 @@ impl NftListStorageOps for IndexedDbNftStorage {
             .await?;
         Ok(())
     }
+
+    async fn get_nfts_by_token_address(&self, chain: Chain, token_address: String) -> MmResult<Vec<Nft>, Self::Error> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let table = db_transaction.table::<NftListTable>().await?;
+
+        let index_keys = MultiIndex::new(CHAIN_TOKEN_ADD_INDEX)
+            .with_value(chain.to_string())?
+            .with_value(&token_address)?;
+
+        table
+            .get_items_by_multi_index(index_keys)
+            .await?
+            .into_iter()
+            .map(|(_item_id, item)| nft_details_from_item(item))
+            .collect()
+    }
+
+    async fn update_nft_spam_by_token_address(
+        &self,
+        chain: &Chain,
+        token_address: String,
+        possible_spam: bool,
+    ) -> MmResult<(), Self::Error> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let table = db_transaction.table::<NftListTable>().await?;
+
+        let index_keys = MultiIndex::new(CHAIN_TOKEN_ADD_INDEX)
+            .with_value(chain.to_string())?
+            .with_value(&token_address)?;
+
+        let nfts: Result<Vec<Nft>, _> = table
+            .get_items_by_multi_index(index_keys)
+            .await?
+            .into_iter()
+            .map(|(_item_id, item)| nft_details_from_item(item))
```

Comment:
please use `.iter()` instead and pass by ref

Reply:
Sorry I didnt get the idea. why I should avoid table/item objects consumption? 
I noticed that I dont use IndexedDB tables anymore after building Nft or NftTransfer objects from them, thats why I dont use ref and clone details_json and prefer to use into_iter to consume tables.

Reply:
because `into_iter()` does involve moving or cloning the elements while `iter()` doesn't... so generally using `iter()` is generally more memory efficient and I think we should use always use `iter()` where possible instead of the `into_iter()`

Reply:
> into_iter() does involve moving or cloning the elements

nft structures dont impl copy, so into_iter just moves ownership, just updates the pointer, without duplicating the items in memory, which is generally efficient for memory. 
Why ownership move isn't memory efficient? I thought it is a general suggestion to move ownership in rust if you dont need objects anymore.
If we use iter, then we have to clone details_json field, so why keeping all items plus cloning details json in moment is better?
I first time see the info that moving ownership is not as good as taking ref and cloning.

Reply:
Okay I understand now that the data aren't used afterwards so no problem. thanks